pyanaconda/installation.py

```diff
@@ -331,14 +331,11 @@ def run_pre_install():
     installation_queue.append(payload_install)
 
     # for some payloads storage is configured after the payload is installed
-    if not payload.needs_storage_configuration:
+    if not payload.needs_storage_configuration and not conf.target.is_directory:
```

Comment:
This was separated in this way for a reason I think...maybe using lorax/LMC to create live rpm-ostree? I forget.  May not be relevant anymore.  `git annotate` would probably know though.

Reply:
Ups you are correct. We would probably broke the --dirinstall by calling the part of this code there.
Thanks a lot for your idea!

Reply:
Also there is a question, would dirinstall without this code be usable. The code creates /var and similar mountpoints so I can't imagine system usable without this. Also is dirinstall even useful for ostree?

What do you think @cgwalters ?

Reply:
So or so behavior is the same as before so I'm merging this. We can fix this later.

Reply:
I was able to find out when the check for dir install was removed and it's this commit:
664ef7b43f9102aa9332d0db5b7d13f8ece436f0

It was removed by Brian and the reason was migration to bind mounts. So this seems as a targeted change which improved the situation. In that case I'm taking this as works as expected.